File: frontend/src/services/MentorService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/mentors";

export const getAllMentors = () => {
    return axios.get(API_URL);
};

export const getMentorById = (id) => {
    return axios.get(`${API_URL}/${id}`);
};

export const addMentor = (mentor) => {
    return axios.post(API_URL, mentor);
};

export const updateMentor = (id, mentor) => {
    return axios.put(`${API_URL}/${id}`, mentor);
};

export const deleteMentor = (id) => {
    return axios.delete(`${API_URL}/${id}`);
};

File: backend/mentor-backend/src/main/java/com/studyecart/mentorbackend/service/MentorService.java
package com.studyecart.mentorbackend.service;

import com.studyecart.mentorbackend.entity.Mentor;

import java.util.List;

public interface MentorService {

    Mentor addMentor(Mentor mentor);

    List<Mentor> getAllMentors();

    Mentor getMentorById(Long id);

    Mentor updateMentor(Long id, Mentor mentor);

    void deleteMentor(Long id);
}